ext/filter/logical_filters.c

```diff
@@ -237,6 +237,10 @@ void php_filter_int(PHP_INPUT_FILTER_PARAM_DECL) /* {{{ */
 				error = 1;
 			}
 		} else if (allow_octal) {
+			/* Support explicit octal prefix notation */
+			if (*p == 'o' || *p == 'O') {
```

Comment:
Out of scope - is the previous behavior of `0x` (and new behavior of `'0o'`) intended for filter_var? It returns 0 but I'd have assumed false from the way octals/hex are represented elsewhere

```php
php > var_export(filter_var('0x', FILTER_VALIDATE_INT, FILTER_FLAG_ALLOW_HEX));
0
php > var_export(filter_var('0xg', FILTER_VALIDATE_INT, FILTER_FLAG_ALLOW_HEX));
false
php > var_export(filter_var('', FILTER_VALIDATE_INT, FILTER_FLAG_ALLOW_HEX));
false
```

Reply:
That's a rather odd behaviour that I didn't think was possible, if I introduced the behaviour in ext-filter I'll work around it but thanks for finding this will add some tests when I get around to this again. :-)

Reply:
As these are now fixed in PHP 8, a string with only an octal string will fail.